dmoj/generator.py

```diff
@@ -6,20 +6,16 @@
 
 
 class GeneratorManager(object):
-    def __init__(self):
-        self._cache = {}
-
-    def get_generator(self, filename, flags):
+    def get_generator(self, filenames, flags):
         from dmoj.executors import executors
 
-        filename = os.path.abspath(filename)
-        cache_key = filename, tuple(flags)
-        if cache_key in self._cache:
-            return self._cache[cache_key]
+        filenames = map(os.path.abspath, filenames)
```

Comment:
Is this even necessary?

Reply:
I actually had my doubts about that too, but the previous code did it so I replicated it for the multi-file scenario.